monai/networks/trt_wrapper.py

```diff
@@ -0,0 +1,441 @@
+                kwargs.update(self._inputs_to_dict(argv))
+                argv = []
+
+            if self.engine is None and not self.disabled:
+                self._build_and_save(kwargs)
+            if self.engine is not None:
+                # forward_trt is not thread safe as we do not use per-thread execution contexts
+                with lock_sm:
+                    device = torch.cuda.current_device()
+                    stream = torch.cuda.Stream(device=device)
+                    self.engine.set_inputs(kwargs, stream.cuda_stream)
+                    self.engine.allocate_buffers(device=device)
+                    # Need this to synchronize with Torch stream
+                    stream.wait_stream(torch.cuda.current_stream())
+                    ret = self.engine.infer(stream.cuda_stream, use_cuda_graph=self.use_cuda_graph)
+                    ret = list(ret.values())
+
+                    if len(ret) == 1:
+                        ret = ret[0]
+                    return ret
+        except Exception as e:
+            if self.model:
+                LOGGER.info(f"Exception: {e}\nFalling back to Pytorch ...")
+            else:
+                raise e
+        return self.model.forward(*argv, **kwargs)
+
+    def _onnx_to_trt(self):
+        """
+        Builds TRT engine from ONNX file at self.onnx_path and saves to self.trt_path
+        """
+
+        profiles = []
+        if self.profiles:
+            for input_profile in self.profiles:
+                if isinstance(input_profile, Profile):
+                    profiles.append(input_profile)
+                else:
+                    p = Profile()
+                    for name, dims in input_profile.items():
+                        assert len(dims) == 3
+                        p.add(name, min=dims[0], opt=dims[1], max=dims[2])
+                    profiles.append(p)
+
+        build_args = self.build_args.copy()
+        build_args["tf32"] = self.precision != "fp32"
+        build_args["fp16"] = self.precision == "fp16"
+        build_args["bf16"] = self.precision == "bf16"
+
+        LOGGER.info(f"Building TensorRT engine for {self.onnx_path}: {self.engine_path}")
+        network = network_from_onnx_path(self.onnx_path, flags=[trt.OnnxParserFlag.NATIVE_INSTANCENORM])
+        engine = engine_from_network(network, config=CreateConfig(profiles=profiles, **build_args))
+        save_engine(engine, path=self.engine_path)
+
+    def _build_and_save(self, input_example):
+        """
+        If TRT engine is not ready, exports self.model to ONNX,
+        builds TRT engine and saves serialized TRT engine to the disk.
+        Args:
+             input_example: passed to onnx.export()
+        """
+        if self.engine is not None:
+            return
+
+        if self.model is None:
+            raise ValueError("ERROR: self.model is None!")
+
+        try:
+            export_args = self.export_args
+            dbs = self.dynamic_batchsize
+            if dbs:
```

Comment:
It looks like users can provide dynamic size more than batchsize here. Is this supported by the TensorRT now?

Thanks  
Bin

Reply:
If user wants to specify additional dynamic shapes, TRTWrapper supports it via 'input profiles' argument - in this case, all axes can be dynamic. And more than one profile can be specified for ultimate flexibility. 
I have added dynamic_batchsize arg for compatibility with existing MONAI APIs. It is indeed a convenient way to set only batchsize as dynamic - which is the case with most visual networks.